ocpmodels/common/relaxation/ml_relaxation.py

```diff
@@ -40,25 +44,45 @@ def ml_relax(
         save_full_traj: bool
             Whether to save out the full ASE trajectory. If False, only save out initial and final frames.
     """
-    batch = batch[0]
-    ids = batch.sid
-    calc = TorchCalc(model, transform)
+    batches = deque([batch[0]])
+    relaxed_batches = []
+    while batches:
+        batch = batches.popleft()
+        oom = False
+        try:
+            ids = batch.sid
+            calc = TorchCalc(model, transform)
 
-    # Run ML-based relaxation
-    traj_dir = relax_opt.get("traj_dir", None)
-    optimizer = LBFGS(
-        batch,
-        calc,
-        maxstep=relax_opt.get("maxstep", 0.04),
-        memory=relax_opt["memory"],
-        damping=relax_opt.get("damping", 1.0),
-        alpha=relax_opt.get("alpha", 70.0),
-        device=device,
-        save_full_traj=save_full_traj,
-        traj_dir=Path(traj_dir) if traj_dir is not None else None,
-        traj_names=ids,
-        early_stop_batch=early_stop_batch,
-    )
-    relaxed_batch = optimizer.run(fmax=fmax, steps=steps)
+            # Run ML-based relaxation
+            traj_dir = relax_opt.get("traj_dir", None)
+            optimizer = LBFGS(
+                batch,
+                calc,
+                maxstep=relax_opt.get("maxstep", 0.04),
+                memory=relax_opt["memory"],
+                damping=relax_opt.get("damping", 1.0),
+                alpha=relax_opt.get("alpha", 70.0),
+                device=device,
+                traj_dir=Path(traj_dir) if traj_dir is not None else None,
+                traj_names=ids,
+                early_stop_batch=early_stop_batch,
+            )
+            relaxed_batch = optimizer.run(fmax=fmax, steps=steps)
+            relaxed_batches.append(relaxed_batch)
+        except RuntimeError as e:
```

Comment:
Probably better to explicitly catch `torch.cuda.OutOfMemoryError` not all `RuntimeError`s

Reply:
OOM is a runtime error, the above is not valid. See whatt fairseq is doing - https://github.com/facebookresearch/fairseq/blob/50a671f78d0c8de0392f924180db72ac9b41b801/fairseq/trainer.py#L283.